Content.Shared/Mind/MindComponent.cs

```diff
@@ -1,84 +1,86 @@
+    ///     of the mind. As a result it should be safe to network "secret" information like roles & objectives
     /// </remarks>
-    [RegisterComponent]
+    [RegisterComponent, NetworkedComponent, AutoGenerateComponentState(true)]
     public sealed partial class MindComponent : Component
     {
-        internal readonly List<EntityUid> Objectives = new();
+        [DataField, AutoNetworkedField]
+        public List<EntityUid> Objectives = new();
 
         /// <summary>
         ///     The session ID of the player owning this mind.
         /// </summary>
-        [ViewVariables, Access(typeof(SharedMindSystem))]
+        [DataField, AutoNetworkedField, Access(typeof(SharedMindSystem))]
         public NetUserId? UserId { get; set; }
 
         /// <summary>
         ///     The session ID of the original owner, if any.
         ///     May end up used for round-end information (as the owner may have abandoned Mind since)
         /// </summary>
-        [ViewVariables, Access(typeof(SharedMindSystem))]
+        [DataField, AutoNetworkedField, Access(typeof(SharedMindSystem))]
         public NetUserId? OriginalOwnerUserId { get; set; }
 
         /// <summary>
-        ///     Entity UID for the first entity that this mind controlled. Used for round end.
+        ///     The first entity that this mind controlled. Used for round end information.
         ///     Might be relevant if the player has ghosted since.
         /// </summary>
-        [ViewVariables] public EntityUid? OriginalOwnedEntity;
+        [DataField, AutoNetworkedField]
+        public NetEntity? OriginalOwnedEntity;
+        // This is a net entity, because this field currently ddoes not get set to null when this entity is deleted.
+        // This is a lazy way to ensure that people check that the entity still exists.
+        // TODO MIND Fix this properly by adding an OriginalMindContainerComponent or something like that.
 
         [ViewVariables]
         public bool IsVisitingEntity => VisitingEntity != null;
 
-        [ViewVariables, Access(typeof(SharedMindSystem))]
+        [DataField, AutoNetworkedField, Access(typeof(SharedMindSystem))]
         public EntityUid? VisitingEntity { get; set; }
 
         [ViewVariables]
         public EntityUid? CurrentEntity => VisitingEntity ?? OwnedEntity;
 
-        [ViewVariables(VVAccess.ReadWrite)]
+        [DataField, AutoNetworkedField, ViewVariables(VVAccess.ReadWrite)]
         public string? CharacterName { get; set; }
 
         /// <summary>
         ///     The time of death for this Mind.
         ///     Can be null - will be null if the Mind is not considered "dead".
         /// </summary>
-        [ViewVariables]
+        [DataField]
         public TimeSpan? TimeOfDeath { get; set; }
 
-        /// <summary>
-        ///     The component currently owned by this mind.
-        ///     Can be null.
-        /// </summary>
-        [ViewVariables] public MindContainerComponent? OwnedComponent;
-
```

Comment:
Was it intended to remove the mind container?

Reply:
Yes, the rationale being that other components can and generally should just be obtained from the EntityUid instead of being  directly stored on a component.

Storing the component on another component can make things slightly faster if its really needed as it avoids an extra component lookup, but it makes everything else more convoluted (e.g., serialization to yaml, networking, and code for updating the owner).

But in this case it's not needed for speed and wasn't really even being used.